backend/uclapi/dashboard/models.py

```diff
@@ -70,3 +75,43 @@ class APICall(models.Model):
 
     class Meta:
         _DATABASE = 'default'
+
+
+class Webhook(models.Model):
+    app = models.OneToOneField(App)
+    url = models.URLField(max_length=1000, blank=True)
+
+    siteid = models.CharField(max_length=40, blank=True)
+    roomid = models.CharField(max_length=160, blank=True)
+    contact = models.CharField(max_length=4000, blank=True)
+
+    last_fired = models.DateTimeField(blank=True, null=True)
+
+    verification_secret = models.CharField(
+        max_length=100,
+        default=generate_secret
+    )
+
+    enabled = models.BooleanField(default=False)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        _DATABASE = 'default'
+
+
+class WebhookTriggerHistory(models.Model):
+    webhook = models.ForeignKey(Webhook)
+    payload = models.CharField(max_length=10000000)
```

Comment:
hmm I think some length constraint _is_ useful here. It stops weird cases where we get like 15GB stored in a single field.

Reply:
is that ever a possibility?

Reply:
I mean in principle it's a failsafe, especially given that we don't control the data source (Estates/ISD does). We should not topple even if something gets messed up upstream.

Additionally, if the buckets aren't set up properly (if for example only one is set up), then running `update_gencache` would trigger a webhook event with all roombookings in history (which should obviously fail). The max_length on this field makes it fail, so that's what we want 👍

Reply:
Sounds good then :)